scripts/build/wrap_binary_in_venv.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python2
+"""
+Create a wrapper for a runnable file which requires a certain virtualenv to
+provide its dependencies. The wrapper implicitly uses the virtual environment,
+so it does not need to activated explicitly.
+"""
+
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from logging import getLogger, Formatter, StreamHandler, INFO, DEBUG
+from os import stat, chmod
+from os.path import abspath
+from stat import S_IXUSR
+from string import Template
+
+LOG = getLogger('VirtualenvWrapper')
+
+msg_formatter = Formatter('[%(levelname)s] - %(message)s')
+log_handler = StreamHandler()
+log_handler.setFormatter(msg_formatter)
+LOG.setLevel(INFO)
+LOG.addHandler(log_handler)
+
+
+def add_executable_permission(path):
+    current_stat = stat(path)
+    chmod(path, current_stat.st_mode | S_IXUSR)
+
+
+def generate_content(binary_path, output_path, virtual_environment_path):
+
+    template_content = '\n'.join([
+        r'#!/bin/bash',
+        r'source "${virtual_environment_path}/bin/activate"',
+        r'exec "${binary_path}" "$$@"'])
+
+    template_vars = {
+        'binary_path': binary_path,
+        'output_path': output_path,
+        'virtual_environment_path': virtual_environment_path}
+
+    LOG.debug('Using template:\n%s\nwith the following substitution:\n%s',
+              template_content, template_vars)
+
+    return Template(template_content).substitute(template_vars)
+
+
+def create_venv_wrapper(binary, output, environment):
+    """
+    Create a runnable file which wraps another runnable file. In addition to
+    forwarding all arguments, the wrapper also activates the virtual
+    environment before running the wrapped runnable file.
+    """
+
+    binary_path = abspath(binary)
```

Comment:
should we use absolute paths to the binary? I'm just thinking about using a relative path would have any benefit or not.

Reply:
Virtualenv has issues with relocatable builds. I have also tried to make the binary path relative, but making that work would require getting the path to the bash script from within the script, which I too much effort (the virtual env is not relocatable anyway, there is little use moving the executable out of its build directory).